### DISCUSSION TITLE
sharing state between multiple components

### DISCUSSION THREAD

**Discussion by acivitillo** [idom-team/idom#561] [Question] [answered]
when building dashboards I need to be able to change other components based on the change of other ones. For instance, I change a dropdown and the table below refreshed based on the new value. Is it possible to observe component state from inside another component? So far all examples I could find are with local state within the component.

**Comment by rmorshea** [comment, marked as answer]
Yes, this is possible. The general strategy is to move state which is shared by two or more components into their nearest common parent. Here's a quick example:

```python
from idom import component, use_state, html, run

@component
def SyncedInputs():
    value, set_value = use_state("")
    return html.p(
        Input("First input", value, set_value),
        Input("Second input", value, set_value),
    )

@component
def Input(label, value, set_value):
    def handle_change(event):
        set_value(event["target"]["value"])

    return html.label(label + " ", html.input({"value": value, "onChange": handle_change}))

run(SyncedInputs)
```

<img width="296" alt="image" src="https://user-images.githubusercontent.com/4041990/147995352-9bdeceba-407e-44f6-94ba-56f448870dd5.png">


Note how the shared state is declared in the `SyncedInputs` component rather than in the `Input` component. If it's not too hard to understand, you should be able to take much of the guidance from [React's documentation on this subject](https://beta.reactjs.org/learn/sharing-state-between-components) and apply it to IDOM with little to no modification.

**Reply by acivitillo** [reply]
Thanks! This explanation makes a lot of sense and indeed conceptually I can follow the react docs quite well.

**Comment by acivitillo** [comment]
I have sent you a PR to add the 2 classic React examples including yours above: https://github.com/idom-team/idom/pull/571
